libass/ass_directwrite.c

```diff
@@ -592,8 +592,15 @@ static char *get_fallback(void *priv, ASS_Library *lib,
 #include "ass_directwrite_info_template.h"
 #undef FONT_TYPE
 
+#define FONT_TYPE IDWriteFont
+#define FAMILY_AS_ARG
+#include "ass_directwrite_info_template.h"
+#undef FONT_TYPE
+#undef FAMILY_AS_ARG
```

Comment:
should we just call `CreateFontFace` and use the `IDWriteFontFace` path instead of awkwardly operating on the `IDWriteFont` with a slightly different interface? Or am I misunderstanding how the API works?

Reply:
`IDWriteFontFace` doesn’t provide any access to font names; only `IDWriteFontFace3` does, and it’s not available before Windows 10.

On the other hand, I forgot over time/didn’t know that it’s this easy to convert an `IDWriteFont` to an `IDWriteFontFace`; thanks for pointing it out. I suspect the original reason this wasn’t done was for performance, back when we loaded all fonts eagerly: I’m guessing the face initialization may read in the whole font file and take a lot of time, as opposed to a metadata cache–backed font object. Now that we load fonts lazily, I suppose this would no longer be a big concern if this was the only blocker.

And `FAMILY_AS_ARG` is a micro-optimization, which could, if desired, be simplified by including the argument either unconditionally (and passing `NULL`) or never (and calling `GetFontFamily` even when we already have a family handle: it’s used only in a presumably low-probability fallback branch, anyway).